src/experimental/safe_html_transform/transforms/exp_safe_html.py

```diff
@@ -293,24 +293,22 @@ def convert(self, orig, data, **kwargs):
         else:
             # append html tag to create a dummy parent for the tree
             html = "<html>%s</html>" % orig
+            parser = etree.HTMLParser()
+            tree = etree.parse(StringIO(html), parser)
+            for element in tree.getiterator():
+                if(element.tag == 'h3' or element.tag == 'h4' or element.tag == 'h5' or element.tag == 'h6' or element.tag == 'div'):
+                    element.tag = 'p'
+                if(element.tag == "html" or element.tag == "body" or element.tag == "script"):
+                    etree.strip_tags(tree, element.tag)
```

Comment:
strip_tags() will remove all instances of the given tags, not just one. So you should only have to call strip_tags() once, instead of calling it for every element you find whilst iterating over the tree. Have a look at:-

 http://lxml.de/api/lxml.etree-module.html#strip_tags

Iterating over each node in the tree and modifying it as you go could well result in odd behavior too, so best to move this.